test/gtest/ucs/test_rcache.cc

```diff
@@ -828,3 +829,65 @@ UCS_TEST_F(test_rcache_stats, hits_slow) {
     munmap(mem2, size1);
 }
 #endif
+
+
+class test_rcache_pfn : public ucs::test {
+public:
+    void test_pfn(void *address, unsigned page_num)
+    {
+        pfn_enum_t ctx;
+        ucs_status_t status;
+        unsigned i;
+
+        status = ucs_sys_enum_pfn((uintptr_t)address, page_num, enum_pfn_cb, &ctx);
+        ASSERT_UCS_OK(status);
+        ASSERT_TRUE(ctx.page.size() == ctx.pfn.size());
+        /* check if all pages are listed */
+        for (i = 0; i < page_num; i++) {
+            EXPECT_TRUE(ctx.page.find(i) != ctx.page.end());
+        }
```

Comment:
avoid O(n^2)
just put all PFNs in the set and expect that set size is equal to num_pages

Reply:
fixed